src/GraphQL.Upload.AspNetCore/UploadGraphType.cs

```diff
@@ -13,49 +11,9 @@ public UploadGraphType()
             Description = "A meta type that represents a file upload.";
         }
 
-        public override object ParseLiteral(IValue value)
-        {
-            if (value is NullValue)
-            {
-                return null;
-            }
-
-            if (value is FormFileValue)
-            {
-                return value.Value;
-            }
-
-            return ThrowLiteralConversionError(value);
-        }
-
         public override object ParseValue(object value)
         {
-            return value;
-        }
-
-        /// <inheritdoc />
-        public override IValue ToAST(object value)
-        {
-            var serialized = Serialize(value);
-            if (serialized is IFormFile formFile)
-            {
-                return new FormFileValue(formFile);
-            }
-
-            if (serialized is null)
-            {
-                return new NullValue();
-            }
-
-            throw new NotImplementedException($"Please override the '{nameof(ToAST)}' method of the '{GetType().Name}' scalar to support this operation.");
-        }
-    }
-
-    internal class FormFileValue : ValueNode<IFormFile>
-    {
-        public FormFileValue(IFormFile value)
-        {
-            Value = value;
+            return value as IFormFile;
```

Comment:
I noticed that all builtin scalars in graphql-dotnet always call `ThrowValueConversionError` if the value doesn't match (for example the [BigIntGraphType](https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/Types/Scalars/BigIntGraphType.cs)). I think we should follow this convention. PR #22 also implements these checks (null check and IFormFile type check). Using a switch case like graphql-dotnet could make this very succinct.

Reply:
Makes sense, I'll try to have a look at it tonight!